src/common/rsvc.c

```diff
@@ -83,10 +84,14 @@ rsvc_client_choose(struct rsvc_client *client, crt_endpoint_t *ep)
 	D_DEBUG(DB_MD, DF_CLI"\n", DP_CLI(client));
 	if (client->sc_leader_known && client->sc_leader_aliveness > 0) {
 		chosen = client->sc_leader_index;
+	} else if (client->sc_tried >= client->sc_ranks->rl_nr) {
```

Comment:
Maybe it could use the sc_next >= rl_nr to check if all replicas has been tried since now sc_next always started from 0?
to avoid extra var here.